src/brsTypes/components/RoSGNode.ts

```diff
@@ -231,6 +325,181 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         },
     });
 
+    /** Adds a new field to the node, if the field already exists it doesn't change the current value. */
+    private addfield = new Callable("addfield", {
+        signature: {
+            args: [
+                new StdlibArgument("fieldname", ValueKind.String),
+                new StdlibArgument("type", ValueKind.String),
+                new StdlibArgument("alwaysnotify", ValueKind.Boolean),
+            ],
+            returns: ValueKind.Boolean,
+        },
+        impl: (
+            interpreter: Interpreter,
+            fieldname: BrsString,
+            type: BrsString,
+            alwaysnotify: BrsBoolean
+        ) => {
+            let defaultValue = this.getDefaultValue(type.value);
+
+            if (
+                defaultValue !== Uninitialized.Instance &&
+                this.get(fieldname) === BrsInvalid.Instance
+            ) {
+                this.set(fieldname, defaultValue, alwaysnotify.toBoolean());
+            }
+
+            return BrsBoolean.True;
+        },
+    });
+
+    /** Adds one or more fields defined as an associative aray of key values. */
+    private addfields = new Callable("addfields", {
+        signature: {
+            args: [new StdlibArgument("fields", ValueKind.Object)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, fields: RoAssociativeArray) => {
+            if (!(fields instanceof RoAssociativeArray)) {
```

Comment:
What's the point of this check?  Doesn't the RoAssociativeArray type in the parameter signature prevent this case from ever occurring?

Reply:
It will for the unit tests but won't prevent actual brightscript code to pass random stuff. Also the box returns false in this case so I'm keeping this for consistency